subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -5249,6 +5281,107 @@ static inline uint8_t phy_upd_ind_recv(struct ll_conn *conn, memq_link_t *link,
 }
 #endif /* CONFIG_BT_CTLR_PHY */
 
+#if defined(CONFIG_BT_CTLR_PERIPHERAL_ISO)
+void event_send_cis_rsp(struct ll_conn *conn)
+{
+	struct node_tx *tx;
+
+	tx = mem_acquire(&mem_conn_tx_ctrl.free);
+	if (tx) {
+		struct pdu_data *pdu = (void *)tx->pdu;
+
+		pdu->ll_id = PDU_DATA_LLID_CTRL;
+		pdu->llctrl.opcode = PDU_DATA_LLCTRL_TYPE_CIS_RSP;
+
+		sys_put_le24(conn->llcp_cis.cis_offset_min,
+			     pdu->llctrl.cis_rsp.cis_offset_min);
+		sys_put_le24(conn->llcp_cis.cis_offset_max,
+			     pdu->llctrl.cis_rsp.cis_offset_max);
+		pdu->llctrl.cis_rsp.conn_event_count =
+			sys_cpu_to_le16(conn->llcp_cis.conn_event_count);
+
+		pdu->len = offsetof(struct pdu_data_llctrl, cis_rsp) +
+				    sizeof(struct pdu_data_llctrl_cis_rsp);
+
+		conn->llcp_cis.state = LLCP_CIS_STATE_IND_WAIT;
+
+		ctrl_tx_enqueue(conn, tx);
+	}
+}
+
+void event_peripheral_iso_prep(struct ll_conn *conn, uint16_t event_counter,
+			       uint32_t ticks_at_expire)
+{
+	if (event_counter == conn->llcp_cis.conn_event_count) {
+		ull_peripheral_iso_start(conn, ticks_at_expire);
+
+		conn->llcp_cis.state = LLCP_CIS_STATE_REQ;
+		conn->llcp_cis.req = conn->llcp_cis.ack;
+	}
+}
+
+static uint8_t cis_req_recv(struct ll_conn *conn, memq_link_t *link,
+			    struct node_rx_pdu **rx, struct pdu_data *pdu)
+{
+	struct pdu_data_llctrl_cis_req *req = &pdu->llctrl.cis_req;
+	struct node_rx_conn_iso_req *conn_iso_req;
+	uint16_t cis_handle;
+	uint8_t err;
+
+	conn->llcp_cis.cig_id = req->cig_id;
+	conn->llcp_cis.framed = req->framed;
+	conn->llcp_cis.c_max_sdu = sys_le16_to_cpu(req->c_max_sdu);
+	conn->llcp_cis.p_max_sdu = sys_le16_to_cpu(req->p_max_sdu);
+	conn->llcp_cis.c_sdu_interval = sys_get_le24(req->c_sdu_interval);
+	conn->llcp_cis.p_sdu_interval = sys_get_le24(req->p_sdu_interval);
+	conn->llcp_cis.cis_offset_min = sys_get_le24(req->cis_offset_min);
+	conn->llcp_cis.cis_offset_max = sys_get_le24(req->cis_offset_max);
+	conn->llcp_cis.conn_event_count =
+		sys_le16_to_cpu(req->conn_event_count);
+
+	/* Acquire resources for new CIS */
+	err = ull_peripheral_iso_acquire(conn, &pdu->llctrl.cis_req, &cis_handle);
+	if (err) {
+		return err;
+	}
+
+	conn->llcp_cis.cis_handle = cis_handle;
+	conn->llcp_cis.state = LLCP_CIS_STATE_RSP_WAIT;
+	conn->llcp_cis.req++;
```

Comment:
```suggestion
	conn->llcp_cis.ack -= 2;
```

`ack` is usually decremented by `2` in ULL context and `req` incremented in steps in Thread context.

This is a temporary LLCP implementation, but as it goes upstream, lets be consistent.

Below is the original concept, but over the years things have changed/broken with regards to its actual required use (may be its not even useful in the current upstream implementation).

Thread:
1. Check if `req == ack`
  1.1. Then, proceed with requesting the procedure
  1.2. Else, procedure is already requested/in-progress, proceed to cache the request if not already cached.
2. Increment `req` by 1.
3. Check if `(state = req - ack) == 1`
4. Then,
  4.1. Update procedure context members
  4.2. Increment `req` by 1 again to register the requested procedure (`req - ack` == 2)
5. Else (race with ISR which decremented `ack` by `2`, making the `state = 3`),
  5.1. Decrement `req` by 1 to restore to its original value,
  5.2. Proceed to cache the Thread API request if not already cached

Reply:
Got it.